arviz/stats/stats.py

```diff
@@ -206,7 +214,16 @@ def compare(
     names = []
     for name, dataset in dataset_dict.items():
         names.append(name)
-        ics = ics.append([ic_func(dataset, pointwise=True, scale=scale)])
+        if len(dataset.log_likelihood.data_vars) > 1 and var_name is None:
+            raise ValueError(
+                (
+                    f"Dataset {name} has multiple variables in its log_likelihood.\n"
+                    "In such cases, the var_name parameter is mandatory."
+                )
+            )
```

Comment:
This is basically the same error in https://github.com/arviz-devs/arviz/blob/main/arviz/stats/stats_utils.py#L426, I think it would be better to modify the error in `get_log_likelihood` so compare, loo and waic all show the same error

Reply:
There is an error check there, but it's not helpful because it has lost the identity of the `InferenceData` object. Having the error here means that we can tell the user *which* of the entries in the `dataset_dict` has multiple observed variables (with `name`, as a I do here).  The individual calls to `get_log_likelihood` don't have access to this information.  If we just use the subsidiary errors, then the poor user has to restart and run in the debugger in order to figure out which of the set of `InferenceData` objects has the problem.  This way they can identify the location of the error immediately.

Possibly we could do what I want by catching and re-signaling *all* errors in the `ic_func`, and raise new ones that add `Error in computing the {ic} for {name}` to the original error string.  Would you prefer that?

Also, slightly OT, I note that the subsidiary functions raise a `TypeError` for this situation, where I had thought of it as a `ValueError`.

Reply:
I have a very bad grasp on the meaning of errors, don't hesitate to change it and use the right one.

I was going to write that I'd rather do "catch->re-raise" than defining a completely new error here. But while thinking about this I realized that passing inferencedatas to compare does not make much sense and limits a lot the possibilities. We should be passing elpddata to compare. For now it's mostly a matter of efficiency, if you want to use compare and plot_elpd for example, you have to compute loo/waic twice, there is no way to reuse the result. However, going forward, the current approach is actually making it impossible to use `compare` with reloo or with IWMM. 

I think we should deprecate passing inferencedata and take only elpddata. The first step would be taking both (like plot_elpd does) and print a warning if inferencedata is passed, then eventually remove it.

This does not need to happen here though, we can open an issue about this and homogenize compare, plot_elpd, plot_khat..., for now catching+re-raising is enough. What do you prefer?

Reply:
I would also prefer we push the elpddata issue to a different PR. This one has dragged on long enough!

I need to figure out what multi-observed inference data to use for tests, and then I think this can be finished.